core/os/android/adb/perfetto.go

```diff
@@ -29,9 +33,25 @@ import (
 	"github.com/google/gapid/core/log"
 )
 
+// DeleteFile deletes a file on the device.
+func (b *binding) DeleteFile(ctx context.Context, traceFile string) error {
+	_, err := b.Shell("rm", "-f", traceFile).Call(ctx)
+	if err != nil {
+		err = errors.New(fmt.Sprintf("Failed to delete %s. Try root", traceFile))
+	}
+	return err
+}
+
 // StartPerfettoTrace starts a perfetto trace on this device.
 func (b *binding) StartPerfettoTrace(ctx context.Context, config *perfetto_pb.TraceConfig, out string, stop task.Signal) error {
+
+	// Silently attempt to delete the old trace in case the user has changed
+	// If the user has changed from shell => root, this is fine. For root => shell,
+	// the root user will have still have to manually delete the old trace if it exists
```

Comment:
Where does the user get the message "You need to delete this manually?"

Reply:
Assuming perfetto failed because it couldn't overwrite the existing trace, the user will now see the message:
```Tracing Failed
   Cause: perfetto_cmd.cc:747      Failed to open /data/misc/perfetto-traces/gapis-trace. If you get permission denied in /data/misc/perfetto-traces, the file might have been created by another user, try deleting it first. (errno: 13, Permission denied)```

I chose to silently attempt to delete the file because it doesn't necessarily mean perfetto will fail if deletion fails, but it does fix an edge case failure where a root owned trace is still on the device.